core/arch/interp.c

```diff
@@ -3744,6 +3744,21 @@ build_bb_ilist(dcontext_t *dcontext, build_bb_t *bb)
             break;
         }
 
+#ifdef X86
+        if (my_dcontext != NULL && instr_is_syscall(bb->instr) &&
+            instr_get_opcode(bb->instr) == OP_int &&
```

Comment:
.data can be written later by temporarily unprotecting the target page (while trying to avoid races) -- there are some helpers for this like SELF_UNPROTECT_DATASEC